### PR TITLE
[NUI] Deprecate FlexContainer class

### PR DESCRIPTION
**Pull request by Seoyeon2Kim**
Signed-off-by: Seoyeon Kim <seoyeon2.kim@samsung.com>

### Description of Change ###
-  FlexContainer can be replaced with FlexLayout.

### API Changes ###
 - ACR: https://code.sec.samsung.net/jira/browse/TCSACR-324

### CODE REVIEW

File: src/Tizen.NUI/src/public/BaseComponents/FlexContainer.cs [hunk 1]
```diff
@@ -27,9 +27,9 @@ namespace Tizen.NUI.BaseComponents
     /// FlexContainer can expand items to fill available free space, or shrink them to prevent overflow.<br />
     /// </summary>
     /// <since_tizen> 3 </since_tizen>
+    [Obsolete("Deprecated in API8, Will be removed in API11, " + "Please use FlexLayout instead!")]
```
Comment (m-macias) [inline src/Tizen.NUI/src/public/BaseComponents/FlexContainer.cs:30]: This second comma is unnecessary. Additionally, I don't see any reason why it couldn't be written as a single line of text.
```suggestion
    [Obsolete("Deprecated in API8, Will be removed in API11. Please use FlexLayout instead!")]
```

File: src/Tizen.NUI/src/public/BaseComponents/View.cs [hunk 2]
```diff
@@ -491,6 +491,7 @@ public string TooltipText
         /// If all items in the container set this property, their sizes will be proportional to the specified flex factor.<br />
         /// </summary>
         /// <since_tizen> 3 </since_tizen>
+        [Obsolete("Deprecated in API8, Will be removed in API11, ")]
```
Comment (m-macias) [inline src/Tizen.NUI/src/public/BaseComponents/View.cs:494]: It shouldn't end with comma. There is no suggestion for an alternative property after deprecation, but it is acceptable.
```suggestion
        [Obsolete("Deprecated in API8, Will be removed in API11.")]
```

File: src/Tizen.NUI/src/public/BaseComponents/FlexContainer.cs (hunk 1 above)
Comment (m-macias) [inline src/Tizen.NUI/src/public/BaseComponents/FlexContainer.cs:30]: According to our deprecation policy when you deprecated API level N, we maintain it in N+1 and remove it in N+2, therefore it should be "Deprecated in API8, will be removed in API10" instead.

Comment (Seoyeon2Kim) [reply src/Tizen.NUI/src/public/BaseComponents/FlexContainer.cs]: Removed the comma and changed as a single line.

File: src/Tizen.NUI/src/public/BaseComponents/View.cs (hunk 2 above)
Comment (Seoyeon2Kim) [reply src/Tizen.NUI/src/public/BaseComponents/View.cs]: Removed the comma and changed as a single line.

File: src/Tizen.NUI/src/public/BaseComponents/FlexContainer.cs (hunk 1 above)
Comment (Seoyeon2Kim) [reply src/Tizen.NUI/src/public/BaseComponents/FlexContainer.cs]: Done.

File: src/Tizen.NUI/src/public/BaseComponents/FlexContainer.cs [hunk 3]
```diff
@@ -27,9 +27,9 @@ namespace Tizen.NUI.BaseComponents
     /// FlexContainer can expand items to fill available free space, or shrink them to prevent overflow.<br />
     /// </summary>
     /// <since_tizen> 3 </since_tizen>
+    [Obsolete("Deprecated in API8, will be removed in API10. Please use FlexLayout instead!")]
     public class FlexContainer : View
     {
-        /// This will be public opened in tizen_5.0 after ACR done. Before ACR, need to be hidden as inhouse API.
         [EditorBrowsable(EditorBrowsableState.Never)]
```
Comment (rabbitfor) [inline src/Tizen.NUI/src/public/BaseComponents/FlexContainer.cs:33]: Can you add explanations of the APIs?
```
/// <summary> ...  </summary>
```

Comment (Seoyeon2Kim) [reply src/Tizen.NUI/src/public/BaseComponents/FlexContainer.cs:34]: Done. Add explanations below.

File: src/Tizen.NUI/src/public/BaseComponents/FlexContainer.cs [hunk 4]
```diff
@@ -46,7 +47,7 @@ public class FlexContainer : View
             Tizen.NUI.Object.GetProperty(flexContainer.swigCPtr, FlexContainer.Property.CONTENT_DIRECTION).Get(out temp);
             return (ContentDirectionType)temp;
         });
-        /// This will be public opened in tizen_5.0 after ACR done. Before ACR, need to be hidden as inhouse API.
+        /// <summary> Property of ContentDirection </summary>
```
Comment (dongsug-song) [inline src/Tizen.NUI/src/public/BaseComponents/FlexContainer.cs:50]: is it "Property of FlexDirection" ? could you check and fix it? ^^; Thank you~

File: src/Tizen.NUI/src/public/BaseComponents/FlexContainer.cs [hunk 5]
```diff
@@ -63,7 +64,7 @@ public class FlexContainer : View
             Tizen.NUI.Object.GetProperty(flexContainer.swigCPtr, FlexContainer.Property.FLEX_DIRECTION).Get(out temp);
             return (FlexDirectionType)temp;
         });
-        /// This will be public opened in tizen_5.0 after ACR done. Before ACR, need to be hidden as inhouse API.
+        /// <summary> Property of ContentDirection </summary>
```
Comment (dongsug-song) [inline src/Tizen.NUI/src/public/BaseComponents/FlexContainer.cs:67]: same as above

File: src/Tizen.NUI/src/public/BaseComponents/FlexContainer.cs [hunk 6]
```diff
@@ -138,6 +139,7 @@ public class FlexContainer : View
         /// Calling member functions with an uninitialized handle is not allowed.
         /// </summary>
         /// <since_tizen> 3 </since_tizen>
+        [Obsolete("Deprecated in API8, will be removed in API10. Please use FlexLayout instead!")]
```
Comment (dongsug-song) [inline src/Tizen.NUI/src/public/BaseComponents/FlexContainer.cs:142]: Great! (b^_^)b
Every time doing deprecated ACR, the Obsolete attribute should be added. Thank you.

File: src/Tizen.NUI/src/public/BaseComponents/FlexContainer.cs (hunk 4 above)
Comment (Seoyeon2Kim) [reply src/Tizen.NUI/src/public/BaseComponents/FlexContainer.cs]: Oh Yes, you're right. Sorry for my mistake :(
Thanks for letting me know!

File: src/Tizen.NUI/src/public/BaseComponents/FlexContainer.cs (hunk 5 above)
Comment (Seoyeon2Kim) [reply src/Tizen.NUI/src/public/BaseComponents/FlexContainer.cs]: Done.

**Pull request merged**